### DISCUSSION TITLE
Display of the maximum possible power regardless of the set limit

### DISCUSSION THREAD

**Discussion by hsmwfgie** [tbnobody/OpenDTU#2427] [Ideas]
Hello, the current situation is that when the inverter's power is limited (limit), all displayed values of the strings (e.g. power, irradiation,...) refer to the set limit. Would it be possible to additionally display the maximum achievable power despite the set limit? In other words, that the maximum possible power value is also displayed as if the inverter was not limited.

**Comment by stefan123t** [comment]
@hsmwfgie no we do not know what that value would be. It mostly depends on the irradiation and temperature of your PV modules.
But the value of irradiation we display is based on the currwntly produced power not the potential if there is no limit / 100%.
We simply can not know the potential power generates once the inverter is limited.

**Comment by Gitsarry** [comment]
Would also like to have the value of possible power beyond set limit, for programming dynamic power limting with PV modules that change their potential over the day (e.g. if module 1 power potential is over limit, while actual power of module 2 at the same time is below limit, increase limit of module 1 and decrease limit of module 2).

**Reply by stefan123t** [reply]
@Gitsarry nothing changed, we can only get the currently produced figure with limits. If you remove the limit in your HA or Node-Red sketch for some minutes, e.g. every hour you would get the optimum of the inverters able to produce. But with limit it only returns the actual power.

If you want to have an indication of how much is possible, you may want to look at some weather forecasts for your region. 
Here in Germany MOSMIX has pretty thorough data from all branches of the German Weather Service DWD.
You can then estimate the possible power generation based on your PV panel direction and sun height over the horizon.
But it will always be only an estimate, because you never know what the PV modules will do e.g. based on shadows or temperature.

**Comment by Gitsarry** [comment]
Thanks stefan123t for your explanation and tips. I like a more fact driven approch and solved my use case with the following algorithm in Home Assistant Automation for each of my two Hoymiles inverters for a combined limit of 800W in total, which is working fine:

State "Inverter power at limit" is defined as inverter production power > (inverter power limit - 10 W)

Permanent default limit for each inverter is 400W. So, each morning the inverters start with this power limit.

1. Wait until this inverter gets to state "power at limit" for at least 1 min.
2. Check if other inverter is in state "power at limit". If yes, go to step 5
3. Increase non-permanent power limit of this inverter by 10 W
4. Set non-permanent power limit of other inverter to (800 W - this inverter limit)
5. Wait 1 min
6. Check if this inverter is still in state "power at limit". If yes, go to step 2
7. Go to step 1
